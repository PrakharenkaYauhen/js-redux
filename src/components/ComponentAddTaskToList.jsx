// ComponentAddTaskToList

import React from 'react';
import PropTypes from 'prop-types';

function ComponentAddTaskToList({
    modalCalendarVision,
    onToggleModal,
}) {

    return (
        <div className='todo__tasks'>
            <div className='modal-button'>
                <button className='modal-button__button' onClick={(e) => onToggleModal(modalCalendarVision, e)}>add a new task</button>
            </div>
        </div>
    )
}

ComponentAddTaskToList.propTypes = {
    modalCalendarVision: PropTypes.bool,
    onToggleModal: PropTypes.func,
}

export { ComponentAddTaskToList };