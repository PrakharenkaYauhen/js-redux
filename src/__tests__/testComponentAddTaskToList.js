// testComponentAddTaskToList

import React from 'react';
import ComponentAddTaskToList from '../components/ComponentAddTaskToList';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tasksBlock = renderer
  .create(<ComponentAddTaskToList />)
  .toJSON();
  expect(tasksBlock).toMatchSnapshot();

  // expect(tasksBlock.onToggleModal).toBeDefined();

  // tasksBlock.onToggleModal();
  // tasksBlock = component.toJSON();
  // expect(tasksBlock).toMatchSnapshot();
});