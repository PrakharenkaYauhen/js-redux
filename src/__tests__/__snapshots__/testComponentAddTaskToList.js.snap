// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="todo__tasks"
  hidden={true}
/>
`;
